lasagne/layers/recurrent.py

```diff
@@ -0,0 +1,1350 @@
+    precompute_input=True, mask_input=None, **kwargs)
+
+    A long short-term memory (LSTM) layer.
+
+    Includes optional "peephole connections" and a forget gate.  Based on the
+    definition in [1]_, which is the current common definition.  The output is
+    computed by
+
+    .. math ::
+
+        i_t &= \sigma_i(x_t W_{xi} + h_{t-1} W_{hi}
+               + w_{ci} \odot c_{t-1} + b_i)\\
+        f_t &= \sigma_f(x_t W_{xf} + h_{t-1} W_{hf}
+               + w_{cf} \odot c_{t-1} + b_f)\\
+        c_t &= f_t \odot c_{t - 1}
+               + i_t\sigma_c(x_t W_{xc} + h_{t-1} W_{hc} + b_c)\\
+        o_t &= \sigma_o(x_t W_{xo} + h_{t-1} W_{ho} + w_{co} \odot c_t + b_o)\\
+        h_t &= o_t \odot \sigma_h(c_t)
+
+    Parameters
+    ----------
+    incoming : a :class:`lasagne.layers.Layer` instance or a tuple
+        The layer feeding into this layer, or the expected input shape.
+    num_units : int
+        Number of hidden/cell units in the layer.
+    ingate : Gate
+        Parameters for the input gate (:math:`i_t`): :math:`W_{xi}`,
+        :math:`W_{hi}`, :math:`w_{ci}`, :math:`b_i`, and :math:`\sigma_i`.
+    forgetgate : Gate
+        Parameters for the forget gate (:math:`f_t`): :math:`W_{xf}`,
+        :math:`W_{hf}`, :math:`w_{cf}`, :math:`b_f`, and :math:`\sigma_f`.
+    cell : Gate
+        Parameters for the cell computation (:math:`c_t`): :math:`W_{xc}`,
+        :math:`W_{hc}`, :math:`b_c`, and :math:`\sigma_c`.
+    outgate : Gate
+        Parameters for the output gate (:math:`o_t`): :math:`W_{xo}`,
+        :math:`W_{ho}`, :math:`w_{co}`, :math:`b_o`, and :math:`\sigma_o`.
+    nonlinearity_out : callable or None
+        The nonlinearity that is applied to the output (:math:`\sigma_h`). If
+        None is provided, no nonlinearity will be applied.
+    cell_init : callable, np.ndarray, theano.shared or TensorVariable
+        Initializer for initial cell state (:math:`c_0').  If a
+        TensorVariable (Theano expression) is supplied, it will not be learned
+        regardless of the value of `learn_init`.
+    hid_init : callable, np.ndarray, theano.shared or TensorVariable
+        Initializer for initial hidden state (:math:`h_0').  If a
+        TensorVariable (Theano expression) is supplied, it will not be learned
+        regardless of the value of `learn_init`.
+    backwards : bool
+        If True, process the sequence backwards and then reverse the
+        output again such that the output from the layer is always
+        from :math:`x_1` to :math:`x_n`.
+    learn_init : bool
+        If True, initial hidden values are learned. If `hid_init` or
+        `cell_init` are TensorVariables then the TensorVariable is used and
+        `learn_init` is ignored for that initial state.
+    peepholes : bool
+        If True, the LSTM uses peephole connections.
+        When False, `W_cell_to_ingate`, `W_cell_to_forgetgate` and
+        `W_cell_to_outgate` are ignored.
```

Comment:
Those arguments don't exist since introducing the `Gate` class. You mean `ingate.W_cell` etc., right?